### ISSUE TITLE
Include sheet names in olevba XLM output.

### ISSUE THREAD

**Issue by kirk-sayre-work** [decalage2/oletools#563]
**Is your feature request related to a problem? Please describe.**
The most recent version of olevba looks like it includes the most of the XLM macro information needed to analyze/emulate the behavior of malicious XLM macro Excel files. The one missing piece is being able to tie the XLM macro cells back to the sheet containing the XLM macros. Knowledge of the sheet is needed since the XLM formulas often read data from cells on the same sheet as the XLM macros.

**Describe the solution you'd like**
In terms of desired output, maybe something as simple as including the sheet name in each of the 'BOUNDSHEET : Sheet Information ...' lines and starting each decompiled XLM chunk of lines with something like 'XLM Sheet Name - NAME' ?

**Comment by decalage2** [comment]
Hi @kirk-sayre-work, I think that should be implemented in plugin_biff, so it's a suggestion for @DidierStevens.
Do you mean that you would like to add an XLM interpreter to ViperMonkey (or another tool)? If so, it means a new parser to handle Excel formulas, and then a model of Excel sheets and workbooks to evaluate the formulas. DissectMalware started something similar: https://twitter.com/DissectMalware/status/1250863006298312706
One issue is that plugin_biff does not output the formulas as they are in Excel, so some adaptation might be needed before parsing. Another solution would be to use other Excel parsers to get the full text of formulas, such as xlrd for Excel 97-2003 files (e.g. the modified version by Heat-Miser: https://twitter.com/H_Miser/status/1248247908819845121) and OpenPyXL (https://openpyxl.readthedocs.io/en/stable/) for Excel 2007+.

**Comment by kirk-sayre-work** [comment]
Yes, I'm considering adding an XLM emulator to ViperMonkey. The control flow and data structure functionality in XLM appears to be limited in comparison to VBA so emulating it hopefully should be less difficult than VBA. Right now I'm thinking about just directly parsing the current plugin_biff XLM output to create XLM_Object (XLM equivalent of VBA_Object) objects and storing them in the Excel sheet/workbook classes I already have defined in ViperMonkey.

I saw the modified XLRD package, but I'm trying to avoid requiring forked versions of packages in ViperMonkey (the current number of tools/packages I have added as requirements is bad enough as it is). Oletools is already a ViperMonkey requirement so basing XLM emulation on it won't add any new ViperMonkey requirements.

**Comment by DissectMalware** [comment]
You might find this project interesting as well: https://github.com/DissectMalware/XLMMacroDeobfuscator 

The XLM grammar might be helpful or the interpreter.
